--- a/engine/mouse.s	2026-10-01 18:54:26.887800796 -0500
+++ b/engine/mouse.s	2026-10-01 18:54:26.900800868 -0500
@@ -377,12 +377,12 @@
 vex_wheelv:
 	uses_d1
 
-	move.l	control(a1),a1
-	move.l	14(a1),d0
 	move.l	vwk_real_address(a0),a2
+	move.l	control(a1),a0		; keep a1 for redirect
+	move.l	14(a0),d0
 	move.l	wk_vector_wheel(a2),d1
 	move.l	d0,wk_vector_wheel(a2)
-	move.l	d1,18(a1)
+	move.l	d1,18(a0)
 
 	used_d1
 	; If no mouse type, the original VDI is called too
